cppcache/src/util/string.hpp

```diff
@@ -38,133 +38,19 @@ constexpr std::codecvt_mode codecvt_mode_native_endian =
     endian::native == endian::little ? std::little_endian
                                      : static_cast<std::codecvt_mode>(0);
 
-inline std::u16string to_utf16(const std::string& utf8) {
-#if defined(_MSC_VER) && _MSC_VER >= 1900
-  /*
-   * Workaround for missing std:codecvt identifier.
-   * https://stackoverflow.com/questions/32055357/visual-studio-c-2015-stdcodecvt-with-char16-t-or-char32-t
-   */
-  auto int16String =
-      std::wstring_convert<std::codecvt_utf8_utf16<int16_t>, int16_t>{}
-          .from_bytes(utf8);
-  return std::u16string(reinterpret_cast<const char16_t*>(int16String.data()),
-                        int16String.size());
-#else
-  return std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t>{}
-      .from_bytes(utf8);
-#endif
-}
+std::u16string to_utf16(const std::string& utf8);
```

Comment:
Again you have removed inline optimizations for something that is heavily used outside of this compilation unit. Without whole program optimization support in the linker we may start seeing a loss here.

Reply:
Our intent is to run a profile pass on the native client and selectively optimize, with documentation.

Reply:
Not keen on the concept of throwing out the baby with the bathwater only to later come back and later selectively test. Without testing these changes you have no idea what the impact of these changes have on the profile of the library.

Reply:
Performance pass is done. See the comment below.